pubtools/_pulp/tasks/garbage_collect.py

```diff
@@ -96,13 +99,32 @@ def clean_all_rpm_content(self):
             LOG.info("No repos found for cleaning.")
             return
         arc_repo = clean_repos[0]
-        deleted_arc = list(arc_repo.remove_content(criteria=criteria))
+
         deleted_content = []
-        for task in deleted_arc:
-            if task.repo_id == "all-rpm-content":
+
+        while True:
+            delete_arc_tasks = list(
+                arc_repo.remove_content(
+                    criteria=criteria, limit=UNASSOCIATE_BATCH_LIMIT
+                )
+            )
+            # The pulp client returns task(s) wrapped in a list
+            arc_tasks = list(
+                filter(lambda x: x.repo_id == "all-rpm-content", delete_arc_tasks)
+            )
+            for task in arc_tasks:
                 for unit in task.units:
                     LOG.info("Old all-rpm-content deleted: %s", unit.name)
                     deleted_content.append(unit)
+
+            # Assuming the client can return multiple (python) tasks.
+            # If all tasks delete less units than the batch limit,
+            # implying all relevant units have been unassociated.
+            if not arc_tasks or not any(
+                [t for t in arc_tasks if len(t[0].units) < UNASSOCIATE_BATCH_LIMIT]
```

Comment:
I don't get the `t[0]` here.  What is the type of t?  Isn't it supposed to be a `Task`?

In the loop just above, we do `for task in arc_tasks` and then access `task.units`.  I don't get how it could be valid to access this as both `task.units` and `task[0].units`.